lighthouse-core/gather/gatherers/image-elements.js

```diff
@@ -308,30 +312,39 @@ class ImageElements extends Gatherer {
 
     /** @type {Array<LH.Artifacts.ImageElement>} */
     const imageUsage = [];
-    const top50Images = Object.values(indexedNetworkRecords)
-      .sort((a, b) => b.resourceSize - a.resourceSize)
-      .slice(0, 50);
+
     await Promise.all([
       driver.sendCommand('DOM.enable'),
       driver.sendCommand('CSS.enable'),
       driver.sendCommand('DOM.getDocument', {depth: -1, pierce: true}),
     ]);
 
+    // Sort (in-place) as largest images descending
+    elements.sort((a, b) => {
+      const aRecord = indexedNetworkRecords[a.src] || {};
+      const bRecord = indexedNetworkRecords[b.src] || {};
+      return bRecord.resourceSize - aRecord.resourceSize;
+    });
+
+    // Don't allow more than 3s of this expensive protocol work
```

Comment:
3s does seem a little low tbh, esp. if that is just 85%; 5s? What would 95/99% be?

Reply:
yah 3s is def arbitrary. :) good question


80pctl: 2.5s
85pctl: 3.0s
90pctl: 4.5s
95pctl: 7.5s
99pctl: 20.0s